arangod/Cluster/ClusterRepairOperations.cpp

```diff
@@ -0,0 +1,741 @@
+    return "MoveShardOperation";
+  }
+  std::string operator()(FixServerOrderOperation const& op) const {
+    return "FixServerOrderOperation";
+  }
+};
+
+std::string cluster_repairs::getTypeAsString(RepairOperation const& op) {
+  return boost::apply_visitor(RepairOperationTypeStringVisitor(), op);
+}
+
+VPackBufferPtr MoveShardOperation::toVpackTodo(
+    uint64_t jobId,
+    std::chrono::system_clock::time_point jobCreationTimestamp) const {
+  std::string const serverId = ServerState::instance()->getId();
+
+  std::string const isoTimeString = getExtendedIsoString(jobCreationTimestamp);
+
+  VPackBuilder builder;
+
+  builder.add(VPackValue(VPackValueType::Object));
+
+  builder.add("type", VPackValue("moveShard"));
+  builder.add("database", VPackValue(database));
+  builder.add("collection", VPackValue(collectionId));
+  builder.add("shard", VPackValue(shard));
+  builder.add("fromServer", VPackValue(from));
+  builder.add("toServer", VPackValue(to));
+  builder.add("jobId", VPackValue(std::to_string(jobId)));
+  builder.add("timeCreated", VPackValue(isoTimeString));
+  builder.add("creator", VPackValue(serverId));
+  builder.add("isLeader", VPackValue(isLeader));
+
+  builder.close();
+
+  return builder.steal();
+}
+
+template <typename T>
+VPackSlice RepairOperationToTransactionVisitor::createSingleValueVpack(T val) {
+  VPackBuilder builder;
+  builder.add(VPackValue(val));
+  VPackSlice slice = builder.slice();
+  _vpackBufferArray.emplace_back(std::move(builder.steal()));
+  return slice;
+};
+
+RepairOperationToTransactionVisitor::ReturnValueT
+RepairOperationToTransactionVisitor::operator()(
+    BeginRepairsOperation const& op) {
+  std::string const distributeShardsLikePath =
+      agencyCollectionId(op.database, op.collectionId) + "/" +
+      "distributeShardsLike";
+  std::string const repairingDistributeShardsLikePath =
+      agencyCollectionId(op.database, op.collectionId) + "/" +
+      "repairingDistributeShardsLike";
+  std::string const replicationFactorPath =
+      agencyCollectionId(op.database, op.collectionId) + "/" +
+      "replicationFactor";
+  std::string const protoReplicationFactorPath =
+      agencyCollectionId(op.database, op.protoCollectionId) + "/" +
+      "replicationFactor";
+
+  VPackSlice protoCollectionIdSlice =
+      createSingleValueVpack(op.protoCollectionId);
+  VPackSlice collectionReplicationFactorSlice =
+      createSingleValueVpack(op.collectionReplicationFactor);
+  VPackSlice protoReplicationFactorSlice =
+      createSingleValueVpack(op.protoReplicationFactor);
+
+  std::vector<AgencyPrecondition> preconditions;
+  std::vector<AgencyOperation> operations;
+
+  if (op.renameDistributeShardsLike) {
+    // assert that distributeShardsLike is set, but
+    // repairingDistributeShardsLike
+    // is not
+    preconditions.emplace_back(AgencyPrecondition{
+        distributeShardsLikePath, AgencyPrecondition::Type::VALUE,
+        protoCollectionIdSlice,
```

Comment:
Is it necessary to end all constructors with `,` in this case?
We do not do it anywhere else.